src/app/wallets/ln-send-payment.ts

```diff
@@ -45,66 +51,80 @@ export const lnInvoicePaymentSendWithTwoFA = async ({
   userId,
   twoFAToken,
   logger,
-}: LnInvoicePaymentSendWithTwoFAArgs): Promise<PaymentSendStatus | ApplicationError> => {
-  const decodedInvoice = await decodeInvoice(paymentRequest)
-  if (decodedInvoice instanceof Error) return decodedInvoice
-
-  const { amount: lnInvoiceAmount } = decodedInvoice
-  if (!(lnInvoiceAmount && lnInvoiceAmount > 0)) {
-    return new LnPaymentRequestNonZeroAmountRequiredError()
-  }
-
-  const user = await UsersRepository().findById(userId)
-  if (user instanceof Error) return user
-  const { username, twoFA } = user
-
-  const twoFACheck = twoFA?.secret
-    ? await checkAndVerifyTwoFA({
-        amount: lnInvoiceAmount,
-        twoFAToken: twoFAToken ? (twoFAToken as TwoFAToken) : null,
-        twoFASecret: twoFA.secret,
+}: LnInvoicePaymentSendWithTwoFAArgs): Promise<PaymentSendStatus | ApplicationError> =>
+  asyncRunInSpan(
+    "app.lnInvoicePaymentSendWithTwoFA",
+    {
+      [SemanticAttributes.CODE_FUNCTION]: "lnInvoicePaymentSendWithTwoFA",
+    },
```

Comment:
I think is not good to add this in app layer, cannot be done in graphql layer?

Reply:
If this is added by the parent then it has to be put everywhere this function is called. Imo the spans that are for the app layer need to be put inside the function call.
But we can discuss this in code review.